template/runtime/src/lib.rs

```diff
@@ -336,6 +337,25 @@ impl<F: FindAuthor<u32>> FindAuthor<H160> for FindAuthorTruncated<F> {
 	}
 }
 
+pub struct RandomnessProvider;
+impl
+	frame_support::traits::Randomness<
+		<Runtime as frame_system::Config>::Hash,
+		BlockNumberFor<Runtime>,
+	> for RandomnessProvider
+{
+	fn random(
+		subject: &[u8],
+	) -> (
+		<Runtime as frame_system::Config>::Hash,
+		BlockNumberFor<Runtime>,
+	) {
+		let output = <Runtime as frame_system::Config>::Hashing::hash(subject);
+		let block_number = frame_system::Pallet::<Runtime>::block_number();
+		(output, block_number)
+	}
+}
```

Comment:
This example is dangerous and we should change it. Suggestion in a comment above.

Reply:
Ignore my comment, this should be fine. The spec explicit says that this should not be used as a true randomness source.

Reply:
Sorry for coming back to this, could we append `pallet_timestamp::Pallet::<Runtime>::now()` to the `subject`.

And also add a note saying `This is just an example, this should not be used as a true randomness source`